services/src/main/java/org/keycloak/authentication/authenticators/browser/CookieAuthenticator.java

```diff
@@ -61,7 +62,13 @@ public void authenticate(AuthenticationFlowContext context) {
                 context.setForwardedInfoMessage(Messages.REAUTHENTICATE);
                 context.attempted();
             } else {
-                int previouslyAuthenticatedLevel = acrStore.getHighestAuthenticatedLevelFromPreviousAuthentication();
+                String flowId = context.getExecution().getParentFlow();
```

Comment:
There is method `getTopLevelFlow()` recently added to `AbstractAuthenticationFlowContext` . Could you please use that one instead of this custom code block?